### DISCUSSION TITLE
Reverse proxy forwarded_allow_ips and proxy_headers

### DISCUSSION THREAD

**Discussion by AdnanAvdagic** [emmett-framework/granian#384] [Q&A]
Hello, at my company we currently run gunicorn + uvicorn on AWS

We also use sqladmin and have an issue with 

"Mixed Content: The page at '<URL>' was loaded over HTTPS, but requested an insecure script '<URL>'. This request has been blocked; the content must be served over HTTPS."

Using gunicorn and uvicorn we just needed to add 

forwarded_allow_ips = "*"
proxy_allow_from = "*"

and then everything would work since we run behind a reverse proxy.

I see no wayto do this with Granian, how would one solve it now?

**Comment by gi0baro** [comment]
Given you run behind a reverse proxy, and Granian doesn't provide those parameters, it should work out of the box, as it should behave like if you allow forwarding from everywhere.

The HTTPS sounds like an application issue, not a server one, as you're probably building URLs without intercepting the forwarded protocol which is different from the serving protocol.

**Reply by AdnanAvdagic** [reply]
From gunicorns config

forwarded_allow_ips
Front-end’s IPs from which allowed to handle set secure headers

proxy_allow_ips
Front-end’s IPs from which allowed accept proxy requests 

The point is that the client will be setting the headers, it's not the server

**Reply by AdnanAvdagic** [reply]
From SQLAdmins own page

Deployment with HTTPS
It is common and useful to deploy your SQLAdmin or FastAPI/Starlette application behind a reverse proxy like Nginx and enable HTTPS on the reverse proxy.

Running the app locally you would not face any issues but with HTTPS enabled behind the reverse proxy you might see errors like this in your browser developer console:


Mixed Content: The page at '<URL>' was loaded over HTTPS, but requested an insecure script '<URL>'. This request has been blocked; the content must be served over HTTPS.
This means the CSS and Javascript files for the Admin were not loaded properly. This is not exactly related to the SQLAdmin but more related to how you are deploying your project.

For example if you are using Uvicorn as your ASGI server you can add the following options to solve this issue:

--forwarded-allow-ips='*'
--proxy-headers
So it would be :


uvicorn <module>:<app> --forwarded-allow-ips='*' --proxy-headers
You can find more information and full docs for this at Uvicorn website here at [running behind nginx](https://www.uvicorn.org/deployment/#running-behind-nginx).

https://aminalaee.dev/sqladmin/cookbook/deployment_with_https/
